Core/Core/BackgroundProcessing/Model/BackgroundProcessingInteractor.swift

```diff
@@ -0,0 +1,64 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2023-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import BackgroundTasks
+
+public struct BackgroundProcessingInteractor {
+    private let scheduler: CoreBGTaskScheduler
+
+    public init(scheduler: CoreBGTaskScheduler) {
+        self.scheduler = scheduler
+    }
+
+    public func register(taskID: String) {
+        let result = scheduler.register(forTaskWithIdentifier: taskID,
```

Comment:
Nitpick, but I think renaming `result` to `isRegistered` or something would increase readability. I had to look up the `BGTaskScheduler` documentation to understand why is it a `Bool`.

Reply:
Valid point, I also like when the code is self-documenting.